Cleanup config-4.4

francois-berder-imgtec opened the pull request:
- Remove any lines which define a driver as a module, replacing them by:
  - Concerning USB gadget, I enabled as built-in USB Mass storage (since this is a choice in the menu, I could not select the rest)
  - Anything else has been changed to built-in
- Remove 802.15.4 support from the kernel
- Remove bluetooth support from the kernel
- Enable kmod-bluetooth and kmod-ieee802154 in default target.mk
- Remove audio as built-in
- Enable features which are enabled in kernel 4.1

Running make kernel_oldconfig removes lines from the configuration file but interrupts the building of the kernel.

francois-berder committed:
Remove modules from default kernel config

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

francois-berder committed:
Remove Bluetooth support in the kernel

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

francois-berder committed:
Remove 802.15.4 support in kernel

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

francois-berder committed:
Enable openwrt modules for bluetooth and ieee802154

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

abhijit-mahajani-imgtec reviewed (commented):
Actually kmod-bluetooth adds the following:

> -------CONFIG_BLUEZ \
> -------CONFIG_BLUEZ_L2CAP \
> -------CONFIG_BLUEZ_SCO \
> -------CONFIG_BLUEZ_RFCOMM \
> -------CONFIG_BLUEZ_BNEP \
> -------CONFIG_BLUEZ_HCIUART \
> -------CONFIG_BLUEZ_HCIUSB \
> -------CONFIG_BLUEZ_HIDP \
> -------CONFIG_BT \
> -------CONFIG_BT_BREDR=y \
> -------CONFIG_BT_DEBUGFS=n \
> -------CONFIG_BT_L2CAP=y \
> -------CONFIG_BT_LE=y \
> -------CONFIG_BT_SCO=y \
> -------CONFIG_BT_RFCOMM \
> -------CONFIG_BT_BNEP \
> -------CONFIG_BT_HCIBTUSB \
> -------CONFIG_BT_HCIBTUSB_BCM=n \
> -------CONFIG_BT_HCIUSB \
> -------CONFIG_BT_HCIUART \
> -------CONFIG_BT_HCIUART_BCM=n \
> -------CONFIG_BT_HCIUART_INTEL=n \
> -------CONFIG_BT_HCIUART_H4 \
> -------CONFIG_BT_HIDP \
> -------CONFIG_HID_SUPPORT=y
> so some of the options are not getting enabled?

Moreover 

in kmod-bluetooth 
CONFIG_BT_HCIBTUSB_BCM=n
where as it was CONFIG_BT_HCIBTUSB_BCM=y here.

In short need to verify Bluetooth functionality properly after this change.

francois-berder committed:
Sync kernel 4.4 configuration with kernel 4.1

Some features enabled in kernel 4.1 were missing in kernel
4.4.
This commit enables as built-in all features enabled in 4.1
present in make kernel_menuconfig of kernel 4.4.

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

francois-berder committed:
Remove audio as built-in from kernel 4.4 configuration

The user selects audio support on Ci40 by selecting
kmod-sound-pistachio-soc in make menuconfig.
It must be enabled it as built-in by default.

Signed-off-by: Francois Berder <Francois.Berder@imgtec.com>

abhijit-mahajani-imgtec commented on target/linux/pistachio/config-4.4:
```diff
@@ -32,6 +32,7 @@ CONFIG_BLK_DEV_BSG=y
 CONFIG_BLK_DEV_DM=y
 CONFIG_BLK_DEV_DM_BUILTIN=y
 CONFIG_BLK_DEV_IO_TRACE=y
+CONFIG_BLK_DEV_LOOP=y
```
This is getting removed if you do make kernel_menuconfig and save & exit. 
I think this is dependent on CONFIG_BLK_DEV that's why its getting removed.

francois-berder-imgtec replied:
removed

abhijit-mahajani-imgtec commented on target/linux/pistachio/config-4.4:
```diff
@@ -319,6 +325,7 @@ CONFIG_LEGACY_PTYS=y
 CONFIG_LEGACY_PTY_COUNT=256
 CONFIG_LIBFDT=y
 CONFIG_LKDTM=y
+CONFIG_LOCKD=y
```
After doing make kernel_menuconfig, CONFIG_LOCKD too getting removed.
This is dependent on CONFIG_NFS_FS which was enabled in config-4.1 originally but later on got removed but accidentally CONFIG_LOCKD option stayed in.

francois-berder-imgtec replied:
removed

abhijit-mahajani-imgtec commented on target/linux/pistachio/config-4.4:
```diff
@@ -157,6 +159,7 @@ CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
 CONFIG_DMADEVICES=y
 CONFIG_DMA_ENGINE=y
+CONFIG_DMA_MAYBE_COHERENT=y
```
This option is not there in https://github.com/CreatorDev/linux/blob/openwrt-4.4.7/arch/mips/Kconfig
under MACH_PISTACHIO

only  DMA_NONCOHERENT is there
may be that' why its getting removed when we do make kernel_menuconfig

francois-berder-imgtec replied:
removed

abhijit-mahajani-imgtec commented:
Verified that usbboot, mmcboot, nandboot works fine after the changes.
